crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -715,6 +715,13 @@ where
                 return Ok(())
             }
 
+            let block_number =
+                self.blockchain.block_number(finalized_block_hash)?.ok_or_else(|| {
+                    Error::Provider(ProviderError::UnknownBlockHash(finalized_block_hash))
+                })?;
```

Comment:
this fetch is redundant because we already do this on L725

we can use `finalized.number` directly

Reply:
Actually (just learning how `Deref` works) wouldn't it be more idiomatic to do `finalized.header.number`?